static/app/components/modals/debugFileCustomRepository/appStoreConnect/index.tsx

```diff
@@ -111,7 +115,11 @@ function AppStoreConnect({
   const [stepOneData, setStepOneData] = useState<StepOneData>({
     issuer: initialData?.appconnectIssuer,
     keyId: initialData?.appconnectKey,
-    privateKey: initialData?.appconnectPrivateKey,
+    privateKey:
+      typeof initialData?.appconnectPrivateKey === 'object'
```

Comment:
@priscilawebdev confused about this type check. Shouldn't `appconnectPrivateKey` always be an object based on the TS type?

Reply:
yes I understand you and think this could be only a primitive value, but after syncing with the backend we agreed on an object when fetching the data. https://github.com/getsentry/sentry/pull/28192#discussion_r697243854

Reply:
if it's an object it means the secret was not changed

Reply:
@priscilawebdev but if `appconnectPrivateKey` is always an object, why are we checking the type?

Reply:
And if not, what are the other types it could be? Shouldn't we update the TS type to reflect the actual types?

Reply:
when saving or updating, the frontend sends a string and not an object 😉

Reply:
@priscilawebdev shouldn't the type of `appconnectPrivateKey` be `{
    'hidden-secret': boolean;
  } | string` then?

Reply:
in this case the `initialData` will be always `undefined` or an `object`, so the types are correct here. I noticed though that the default value has to be always `undefined`. I've updated the code

Reply:
in this file I've defined the type `string` https://github.com/getsentry/sentry/blob/358548b73c18b5de3a9f9a268add2edd66872320/static/app/components/modals/debugFileCustomRepository/appStoreConnect/types.tsx#L15